ztrouy/cpyou
# Authorization Rework

### Purpose
To no longer require prop drilling when passing authorization to child components using a new authorization context. Also to no longer require encasing all routes that required a User to be logged in with AuthorizedRoute elements

### Files Changed
- Created file, and created the context used for authorization in new `authorizationContext.jsx` component module
- Created file, and created state and context provider for the authorization context in new `authorizationProvider.jsx` component module
- Created file, and wrote import for other modules to only need one import when getting the authorization context in new `useAuthorizationProvider.js` module
- Setup authorization provider for whole application in `main.jsx`
- Removed previous authorization declaration, and use new authorization context in `App.jsx`
- Updated to use athorization context, and updated return to provide an outlet for routes, rather than requiring children to be wrapped in an AuthorizedRoute element in `AuthorizedRoute.jsx`
- Updated to use new authorization context and route authorization, and updated function syntax to use arrow notation in `ApplicationViews.jsx`
- Updated to use new authorization context in `BuildDetails.jsx`, `BuildForm.jsx`, `Comment.jsx`, `Login.jsx`, `Register.jsx`, and `NavBar.jsx`
- Deleted `App.css` and `index.css`

### To Test
Fetch, setup, and run according to project README, then confirm the site still functions as expected without errors
